src/condor_utils/file_transfer.cpp

```diff
@@ -4924,9 +4924,28 @@ FileTransfer::uploadFileList(
 		saved_priv = set_priv( desired_priv_state );
 	}
 
+	//
+	// It is arguably the caller's responsibility to make sure that they
+	// don't request impossible things, or maybe the responsibility of
+	// the FileItem constructor(s) and mutators to avoid allowing such a
+	// thing to be represented, but it seems a lot safer to do this
+	// sanity check here: at least for now, plug-ins can't handle directories.
+	//
+	// (Of course, it would arguably be better not to generate such entries
+	// in the first place, but that's scary for other reasons.)
+	//
+
 	*total_bytes_ptr = 0;
 	for (auto &fileitem : filelist)
 	{
+		if( fileitem.isDirectory() && (! fileitem.destUrl().empty()) ) {
```

Comment:
Can we replace this code with something like

`std::erase_if(filelist,isDirectoryURL)` 

And move it outside the loop?  That way we have less code in the loop, and we might even be able to share the predicate with code higher up in this file.

We can also pull out the `skip_files` skipping outside this loop in the same way.  Unfortunately, I bet that `skip_files` is not guaranteed to be sorted in the same order as `filelist`, so we can't just do the simple and faster `set_difference` on the two vectors.

Reply:
I'm deliberately skipping the entries rather than deleting them on the basis that current discussions with Pelican may result in us wanting to keep them.

I guess since we're copying `checkpointList` into `fileList`, it should be safe to modify `fileList`.

Reply:
If we really want to keep them in the list, we could `std::partition` them to the end of the list, and run the main loop from `begin()` to the partition point?